instrumentation-genai/opentelemetry-instrumentation-google-genai/src/opentelemetry/instrumentation/google_genai/tool_call_wrapper.py

```diff
@@ -0,0 +1,190 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import functools
+import inspect
+from typing import Any, Callable, Optional, Union
+
+from google.genai.types import (
+    ToolListUnion,
+    ToolListUnionDict,
+    ToolOrDict,
+)
+
+from opentelemetry import trace
+from opentelemetry.semconv._incubating.attributes import (
+    code_attributes,
+)
+
+from .flags import is_content_recording_enabled
+from .otel_wrapper import OTelWrapper
+
+ToolFunction = Callable[..., Any]
+
+
+def _to_otel_value(python_value):
+    """Coerces parameters to something representable with Open Telemetry."""
+    if python_value is None:
+        return None
+    if isinstance(python_value, list):
+        return [_to_otel_value(x) for x in python_value]
+    if isinstance(python_value, dict):
+        return {
+            key: _to_otel_value(val) for (key, val) in python_value.items()
+        }
+    if hasattr(python_value, "model_dump"):
+        return python_value.model_dump()
+    if hasattr(python_value, "__dict__"):
+        return _to_otel_value(python_value.__dict__)
+    return repr(python_value)
+
+
+def _create_function_span_name(wrapped_function):
+    """Constructs the span name for a given local function tool call."""
+    function_name = wrapped_function.__name__
+    return f"tool_call {function_name}"
```

Comment:
https://github.com/open-telemetry/semantic-conventions/blob/v1.33.0/docs/gen-ai/gen-ai-spans.md#execute-tool-span seems appropriate

Reply:
Done. Renamed from `tool_call {function_name}` to `execute_tool {function_name}` and applied the other aspects of this specification.